docs/architecture/blockchain.md

```diff
@@ -0,0 +1,70 @@
+
+<p style="text-align: center;">
+    <img src="../img/architecture/blockchain/execution.png" style="width:70%;" alt="Execution diagram"/>
+</p>
+
+## Batch production
+
+A Miden block consist of multiple transaction batches. This enables recursive proving and also allows transactions to be processed concurrently as batches with no overlap in accounts and notes can be built in parallel. Miden will have multiple batch producers operating simultaneously and, together with offchain/client-side transaction proving, this massively reduces the work the network is required to do.
+
+The purpose of this scheme is to produce a single proof that attests to the validity of a number of transactions. This is achieved by recursively verifying each transaction proof within the Miden VM.
+
+<p style="text-align: center;">
+    <img src="../img/architecture/blockchain/batching.png" style="width:50%;" alt="Batch diagram"/>
+</p>
+
+The batch producer aggregates transactions sequentially by verifying their proofs and state transitions are correct. More specifically, the batch producers ensures:
+
+1. **Ordering of transactions**: If several transactions within the same batch affect a single account, the correct ordering must be enforced. For example, if `Tx1` and `Tx2` both describe state changes of account `A`, then the batch kernel must verify them in the order: `A -> Tx1 -> A' -> Tx2 -> A''`.
+2. **Prevention of double spending and duplicate notes**: The batch producer must ensure the uniqueness of all notes across transactions in the batch. This prevents double spending and avoids the situation where duplicate notes, which would share identical nullifiers, are created. Only one of such duplicate notes can later be consumed, as the nullifier will be marked as spent after the first consumption.
+3. **Expiration windows**: It is possible to set an expiration window for transactions, which in turn sets an expiration window for the entire batch. For instance, if transaction `A` expires at block `8` and transaction `B` expires at block `5`, then the batch expiration will be set to the minimum of all transaction expirations, which is `5`.
+
+## Block production
+
+To create a `Block`, multiple batches and their respective proofs are aggregated. `Block` production is not parallelizable and must be performed by the Miden operator. In the future, several Miden operators may compete for `Block` production. The schema used for `Block` production is similar to that in batch production—recursive verification. Multiple batch proofs are aggregated into a single `Block` proof. In addition, the `Block` contains:
+- The commitments to the current global [state](state.md).
+- The newly created nullifiers.
+- The new state commitments for affected private accounts.
+- The full states for all affected public accounts and newly created notes.
```

Comment:
I would also mention the commitments to created notes, to make it clear how private notes are committed to because they are not otherwise mentioned here:

> - The commitments to the newly created notes.